core/arch/arm/kernel/trace_ext.c

```diff
@@ -28,20 +28,39 @@
 #include <trace.h>
 #include <console.h>
 #include <kernel/thread.h>
+#include <kernel/spinlock.h>
+#include <mm/core_mmu.h>
 
 const char trace_ext_prefix[] = "TEE-CORE";
 int trace_level = TRACE_LEVEL;
+static unsigned int puts_lock = SPINLOCK_UNLOCK;
 
 void trace_ext_puts(const char *str)
 {
+	uint32_t itr_status = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
```

Comment:
Only foreign interrupts are masked here.
If some native interrupt or asynchronous abort exception is fired on the executing CPU, the related OP-TEE handler will deadlock if trying to output a trace. Maybe we should better mask all exceptions here.